geometry/optimization/iris.cc

```diff
@@ -522,6 +530,495 @@ std::vector<int> revolute_joint_indices(const multibody::Joint<double>& joint) {
 
 }  // namespace
 
+namespace {
+
+int FindCollisionPairIndex(const MultibodyPlant<double>& plant, 
+Context<double>* context, const Eigen::VectorXd& configuration, 
+const std::vector<GeometryPairWithDistance>& sorted_pairs) {
+  // Call ComputeSignedDistancePairClosestPoints for each pair of collision geometries until 
+  // finding a pair that is in collision and returning the corresponding index
+  int pair_in_collision = -1;
+  int i_pair = 0;
+  for (const auto& pair : sorted_pairs) {
+    plant.SetPositions(context, configuration);
+    auto query_object =
+        plant.get_geometry_query_input_port().template Eval<QueryObject<double>>(*context);
+    const double distance =
+    query_object.ComputeSignedDistancePairClosestPoints(pair.geomA, pair.geomB)
+        .distance;
+    if (distance < 0.0) {
+      pair_in_collision = i_pair;
+      break;
+    }
+    ++i_pair;
+  }
+
+  return pair_in_collision;
+}
+
+std::pair<Eigen::VectorXd, int> CollisionLineSearch(const MultibodyPlant<double>& plant, 
+const Parallelism& parallelism, const planning::CollisionChecker& checker, 
+Context<double>* context, const std::vector<GeometryPairWithDistance>& sorted_pairs, 
```

Comment:
This context can be removed altogether once SceneGraphCollisionChecker returns the pair